src/com/spike/TypeChecker.java

```diff
@@ -0,0 +1,528 @@
+package com.spike;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+class TypeChecker implements ExprVisitor<Integer>, StmtVisitor<Void> {
```

Comment:
Although it's a Java testing implementation, it might be better to group all the type constants in one place (like `static` variables)

Reply:
Good point

Reply:
Did it. Now it should be clearer now.